lib/fauna/query.rb

```diff
@@ -457,13 +457,18 @@ def self.not(boolean)
 
   private
 
+    @auto_var_names = []
+
     def self.with_auto_vars(n_vars)
       low_var_number = Thread.current[:fauna_lambda_var_number] || 0
       next_var_number = low_var_number + n_vars
 
       Thread.current[:fauna_lambda_var_number] = next_var_number
 
-      yield (low_var_number...next_var_number).map { |i| "auto#{i}" }
+      vars = (low_var_number...next_var_number).map do |i|
+        @auto_var_names[i] ||= :"auto#{i}"
```

Comment:
It's possible that the cache will get written to twice, but so long as symbol creation is thread safe, I don't think that should matter? We could always make `@auto_var_names` thread local, but I'm not sure if we have to.